proxy/bootstrap/src/main/java/org/apache/shardingsphere/proxy/arguments/BootstrapArguments.java

```
@@ -97,4 +118,13 @@ private String paddingWithSlash(final String pathArg) {
         }
         return result.toString();
     }
+    
+    private boolean isValidPath(final String path) {
+        try {
+            Paths.get(path);
+        } catch (InvalidPathException | NullPointerException ex) {
+            return false;
```

My understanding is that if user specified an invalid path, the socket will not be present and no error will occur. Am I right?

---

yes, it was designed that way.

---

of course, we can throw a error in this.

---

I prefer throwing an exception if path was invalid.